tuning/sft_trainer.py

```diff
@@ -123,6 +128,16 @@ def train(
     ):
         raise ValueError("gradient_accumulation_steps has to be an integer >= 1")
 
+    if (
+        attention_and_distributed_packing_config is not None
+        and attention_and_distributed_packing_config.padding_free is not None
+        and model_args.use_flash_attn is False
+    ):
+        raise ValueError(
+            "`--padding_free` argument was called without enabling \
+            flash attention, ensure `use_flash_attn = True` to use padding-free flash attention"
+        )
```

Comment:
Lets also have a `packing=False` check when `padding_free` is activated. Put a comment to explain that when using `padding_free`, then we should not attempt to perform any packing using `sft_trainer`